### ISSUE TITLE
toHaveAccessibilityValue always truthy

### ISSUE THREAD

**Issue by gkartalis** [testing-library/jest-native#150]
- `react-native`
- `@testing-library/react-native` version: 11.5.0
- `jest-preset`: "react-native"
- `react-native` version: 0.66.5
- `node` version: 16.18.0

### Relevant code or config:

```js

render(<View testID="view" accessibilityValue={{ text: 'Almost full', disabled: true }} />);

const view = screen.getByTestId('view');


expect(view).toHaveAccessibilityState({ text: 'Almost full', disabled: true }); // passes as expected
expect(view).toHaveAccessibilityState({ text: 'Almost full', disabled: true }); // fails as expected

expect(view).toHaveAccessibilityValue({ text: "Almost full" }); // passes as expected
expect(view).toHaveAccessibilityValue({ text: "Whatever this is wrong" }); // passes but it shouldn't pass!


```

### What you did:

Was trying to make some assertions on a single a11y value on a test (in this project https://github.com/artsy/eigen) and noticed the behavior that I mention on the Relevant code or config part.

### What happened:

The first assertion was passing as expected 
```js
expect(view).toHaveAccessibilityValue({ text: "Almost full" }); // passes as expected
```

but then I was like, lets make the test fail to make sure it works properly

```js
expect(view).toHaveAccessibilityValue({ text: "Whatever this is wrong" }); // passes but it shouldn't pass!
```

and noticed the above.

### Reproduction:

I think that it can be reproduced really easily in any project by copying the Relevant code part

### Problem description:

toHaveAccessibilityValue should make the test fail if no single match is found

### Suggested solution:

not sure yet but will dive deeper

### Can you help us fix this issue by submitting a pull request?

Would love to help fixing this

**Comment by mdjastrzebski** [comment]
@gkartalis not sure what exactly is happening at your side, so I'll do my best to address your issue.

I've spotted few errors in your test code:
1. `disabled: true` should be put inside `accessibilityState` and not `accessibilityValue` prop for `View`
2. First two `expect` (with `accessibilityState`) are also putting `text` in `toHaveAccessibilityState` which should be put in `toHaveAccessibilityView`. They are also the same, despite comments about one failing and other not.

The following test passes correctly on my side:
```ts
test('issue', () => {
  render(
    <View
      testID="view"
      accessibilityValue={{ text: 'Almost full' }}
      accessibilityState={{ disabled: true }}
    />
  );

  const view = screen.getByTestId('view');

  expect(view).toHaveAccessibilityState({ disabled: true });
  expect(view).not.toHaveAccessibilityState({ disabled: false });

  expect(view).toHaveAccessibilityValue({ text: 'Almost full' });
  expect(view).not.toHaveAccessibilityValue({ text: 'Whatever this is wrong' });
});
```

Let me know if correcting the spotted issues resolves the error for you. If not then pls prepare a minimal repro repository based on [RNTL basic example](https://github.com/callstack/react-native-testing-library/tree/main/examples/basic), so that I can reproduce the issue locally.

**Comment by gkartalis** [comment]
Gotcha, honestly I went ahead and thought that`toHaveAccessibilityValue` is the same as `toHaveAccessibilityState` but partial for a single state value. That was also the reason that my example was like that.

I can now close the issue then, but the thing is that `toHaveAccessibilityValue` was not failing even though I didn't have any accessibilityValue prop included 🤔, any way closing the issue.

Thanks for the response!

**Comment by gkartalis** [comment]
Good point good point it was an unfortunate typo :D . But thanks for the links (I will also edit the initial post to reflect that)
